server/expression.js

```diff
@@ -357,12 +373,15 @@ expression.readMatrix_ = function(fileName, inputGenes,
     });
   }
   geneNames.forEach(function(geneName) {
-    var geneIndex = allGeneNames[geneName];
-    if (geneName in allGeneNames) {
+    var geneIndex = allGeneNames[geneName.toLowerCase()].index;
+    if (geneName.toLowerCase() in allGeneNames) {
       var parts = lines[geneIndex].split(/[\t\s]+/);
       var tmpLine = [];
       conditions.forEach(function(conditionIndex) {
         var value = parseFloat(parts[conditionIndex]);
+        if (!value) {
```

Comment:
That sounds reasonable. A few concerns are
* If there is an error in the data, we may want to notify the user, instead of just ignoring part of the data.
* We may want to consider validating the uploaded data, before we store them to the disk. That validation could be added after uploading.